src/lib-python/ioda_conv_ncio.py

```diff
@@ -328,11 +347,19 @@ def ExtractObsData(self, ObsData):
             VarMap[VarList[i]] = i
 
         # Preallocate arrays and fill them up with data from the dictionary
-        ObsVars = {
-            self._oval_name  : np.full((self._nvars, self._nlocs), self._defaultF4),
-            self._oerr_name  : np.full((self._nvars, self._nlocs), self._defaultF4),
-            self._oqc_name   : np.full((self._nvars, self._nlocs), self._defaultI4),
-            }
+	ObsVars = {}
+	for o in range(len(ObsVarList)):
+            VarType = type(ObsVarExamples[o])
+            if (VarType in [float,np.float32,np.float64]):
+                defaultval = self._defaultF4
+            elif (VarType in [int,np.int64,np.int32]):
+                defaultval = self._defaultI4    # convert long to int
+            elif (VarType in [str,np.s1]):
```

Comment:
np.s1 does not exist

Reply:
yeah sorry it should be np.str_ I didn't realize this one didn't follow the convention of the others